rnd/autogpt_builder/src/constants/constants.ts

```diff
@@ -0,0 +1,5 @@
+export const STATUS_INCOMPLETE = 'INCOMPLETE';
```

Comment:
These are not application constants; on the server side it's an `enum` type. They are part of the API and imo should be stored with the rest of the API code.

Remember this is a TypeScript project, and TypeScript supports type specifications like:

```ts
type FlowExecutionResult = {
  /* ... */
  status: "INCOMPLETE" | "QUEUED" | "RUNNING" | "COMPLETED" | "FAILED"
  /* ... */
}

const status: FlowExecutionResult['status'] = "QUEUED"
```
* No need to store a type separately to use it in a type hint somewhere
* Attributes with a finite set of possible values are easy to define 
* TypeScript will enforce that a valid value is assigned
* IDE will provide autocomplete hints based on the type definition

Reply:
I think we can split constants as
constants/api
constants/something_else

the overall concept of

api
components
styles
types
hooks

etc
seems to be the nextjs way - and imo I find it quite logical

and then we can split into further areas such as api, like you said with the constants ones

Reply:
> They are part of the API and imo should be stored with the rest of the API code

they are part of the API but are also quite general, and will most likely be used within the application beyond just being fetched from the API

Reply:
> they are part of the API but are also quite general, and will most likely be used within the application beyond just being fetched from the API

Allow me to rephrase: the types are *dictated by* the API. They originate from the API. So they should be defined as part of the API, not outside of it.